src/include/duckdb/main/capi/header_generation/header_base.hpp.template

```diff
@@ -878,6 +878,17 @@ typedef struct _duckdb_catalog_entry {
 	void *internal_ptr;
 } * duckdb_catalog_entry;
 
+//===--------------------------------------------------------------------===//
+// Logging Types
+//===--------------------------------------------------------------------===//
+
+//! Holds a log storage object.
+typedef struct _duckdb_log_storage {
+	void *internal_ptr;
+} * duckdb_log_storage;
+
+typedef void (*duckdb_logger_write_log_entry_t)(duckdb_timestamp timestamp, const char *level, const char *log_type, const char *log_message);
```

Comment:
Maybe we can add a comment here that in future versions we might break this / expose a second type to also allow passing the log context?